src/cbor_common.c

```diff
@@ -15,12 +15,14 @@ _Static_assert((sizeof(size_t) == sizeof(void *)),
 
 bool new_backup(cbor_state_t *state, uint32_t new_elem_count)
 {
-	if ((state->backups->current_backup + 1)
-		>= state->backups->num_backups) {
+	if (!state->backups || ((state->backups->current_backup)
+		>= state->backups->num_backups)) {
 		FAIL();
 	}
 
-	uint32_t i = ++(state->backups->current_backup);
+	/* use the backup at current_backup - 1, since otherwise, the 0th
```

Comment:
Is this comment correct? It seems like we are simply post-incrementing, not using the backup at `current_backup - 1` as the comment states.

Reply:
I rearranged the code to make more sense, take a look.